test/e2e/webhook_test.go

```diff
@@ -460,3 +463,145 @@ var _ = Describe("Fleet's Hub cluster webhook tests", func() {
 		})
 	})
 })
+
+var _ = Describe("Fleet's CRD Resource Handler webhook tests", func() {
```

Comment:
Hi Arvind! Just a few nits:

* There are many repeated constants in the code right now (e.g, `test-user`, etc.); it could help if they are extracted out
* Though there's nothing wrong comparing error messages, it might help to check against [status codes (reasons)](https://pkg.go.dev/k8s.io/apimachinery@v0.27.2/pkg/apis/meta/v1#StatusReason)
* Style-wise, gomega has a To(Succeed()) matcher, which might be simpler than checking for nil errors.